recipes/gf-complete/all/conanfile.py

```diff
@@ -61,44 +66,43 @@ def _configure_autotools(self):
             self, win_bash=tools.os_info.is_windows)
 
         with tools.environment_append(self._autotools.vars):
-            # run autoreconf
-            self.run("./autogen.sh", win_bash=tools.os_info.is_windows)
+            with tools.chdir(self._source_subfolder):
+                self.run("./autogen.sh", win_bash=tools.os_info.is_windows)
 
         if "x86" in self.settings.arch:
             self._autotools.flags.append('-mstackrealign')
 
-        if "fPIC" in self.options:
-            self._autotools.fpic = self.options.fPIC
-
         configure_args = [
             "--enable-shared=%s" % ("yes" if self.options.shared else "no"),
             "--enable-static=%s" % ("no" if self.options.shared else "yes")
         ]
 
-        # enabled by default
-        if not self.options.neon:
-            configure_args.append("--disable-neon")
-        if not self.options.sse:
-            configure_args.append("--disable-sse")
+        if "arm" in self.settings.arch:
+            # neon is enabled by default
+            if self.options.neon != "auto" and not self.options.neon:
+                configure_args.append("--disable-neon")
```

Comment:
where is the True/enable_neon? I only see condition for auto and False.

Reply:
neon and sse are enabled by default in the gf-complete build system, avx is disabled by default. The build system only offers `--disable-neon`, `--disable-sse` and `--enable-avx` (and `--enable-valgrind`, which I didn't implement in the package).

Thus I only use these configure parameters, if the options are set accordingly.

There is a generic `--disable/enable-FEATURE`, but I don't think be used here. The three specific options for `--disable-neon`, `--disable-sse` and `--enable-avx` are explicitly handled in `configure.ac` and trigger some flags to be set ([see here](https://github.com/ceph/gf-complete/blob/a6862d10c9db467148f20eef2c6445ac9afd94d8/configure.ac#L42
)). The generic `--disable/enable-FEATURE` parameters will probably not do that.

It's a bit confusing and I hoped the `# ... is enabled/disabled by default` comments might help. Do you think I should add an `else: pass  # ...` case to make it more clear?

Reply:
I see, but I prefer being explicit, it helps for maintenance and reading in the future. This recipe is not the only one which uses "auto" option. Please, set the arguments according the options:

```
    if self.options.neon:
        configure_args.append("--enable-neon")
     elif not self.options.neon:
        configure_args.append("--disable-neon")
```

Reply:
I understand. It certainly improves readability. I thought we'd want to avoid setting "invalid" options in the build system, but since they are simply ignored, I guess it's fine.

I followed the example of the dlib recipe you linked above.